### PR TITLE
Improves the hasScore for the assessmentresult.

### PR DESCRIPTION
**Pull request by terw-dan**
Adds a boolean isScoreSet to the assessmentResult. This way we can check if a score is set, and it is possible to set a score to 0.

### CODE REVIEW

File: js/values/AssessmentResult.js [hunk 1]
```diff
@@ -29,14 +29,15 @@ AssessmentResult.prototype.getScore = function() {
 
 /**
  * Set the score for the assessment.
- * @param {number|string} score The score to be used for the score property
+ * @param {number} score The score to be used for the score property
  * @returns {void}
  */
 AssessmentResult.prototype.setScore = function( score ) {
-	if ( !isNumber( score ) && !isString( score ) ) {
+	this.isScoreSet = true;
```
Comment (atimmer) [inline js/values/AssessmentResult.js]: Why not?

```js
if ( !isNumber( score ) ) {
    this.score = score;
    this.isScoreSet = true;
}
```

File: js/values/AssessmentResult.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 var isUndefined = require( "lodash/isUndefined" );
 var isNumber = require( "lodash/isNumber" );
-var isString = require( "lodash/isString" );
 
 /**
  * Construct the AssessmentResult value object.
  * @constructor
  */
 var AssessmentResult = function() {
+	this.isScoreSet = false;
```
Comment (atimmer) [inline js/values/AssessmentResult.js]: Why not call this `hasScore`, the method is called this way.

**Comment by atimmer** [comment]
CR Done 👷🏻

File: js/values/AssessmentResult.js (hunk 1 above)
Comment (terw-dan) [inline js/values/AssessmentResult.js]: changed this to 
if ( isNumber( score ) ) {
    this.score = score;
    this.isScoreSet = true;
}

File: js/values/AssessmentResult.js (hunk 2 above)
Comment (terw-dan) [inline js/values/AssessmentResult.js]: renamed it to _hasScore :)

**Comment by atimmer** [comment]
CR 2 & Acceptance done 👍🏻

**Pull request merged**